File: src/main/java/com/rcoverzola/ejers/SumAllNum.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.rcoverzola.ejers;

import com.util.ProgramUtil;

/**
 *
 * @author Roger Coverzola Bogado
 */
public class SumAllNum extends ProgramUtil {

  private static int ZERO = 0;

  public SumAllNum() {
    String[] message = {"Ingrese numeros enteros positivo, ó ingesar un numero negativo para salir: ", "El resultado de la suma es: ", "Adios!!"};
    init(message);
  }

  public void cal() {
    int aux = 0;
    while (super.EXIT) {
      System.out.println(super.MESSAGES[0]);
      Integer value = super.scanner.nextInt();
      if (value < this.ZERO) {
        super.EXIT = false;
      } else {
        aux += value;
      }
    }
    System.out.println(super.MESSAGES[1] + aux);
    System.out.println(super.MESSAGES[2]);
  }

}

File: src/SumAllNum.ts
import { ProgramUtil } from './ProgramUtil';

export class SumAllNum extends ProgramUtil {
    private static readonly ZERO: number = 0;

    constructor() {
        super();
        const message: string[] = [
            "Ingrese numeros enteros positivo, ó ingesar un numero negativo para salir: ", 
            "El resultado de la suma es: ", 
            "Adios!!"
        ];
        this.init(...message);
    }

    public cal(): void {
        let aux: number = 0;
        
        const askNumber = () => {
            if (!this.EXIT) {
                console.log((this.MESSAGES[1] || '') + aux);
                console.log(this.MESSAGES[2]);
                this.scanner.close();
                return;
            }
            
            console.log(this.MESSAGES[0]);
            
            this.scanner.question('', (input: string) => {
                const value: number = parseInt(input);
                
                if (value < SumAllNum.ZERO) {
                    this.EXIT = false;
                } else {
                    aux += value;
                }
                
                askNumber();
            });
        };
        
        askNumber();
    }
}
